File: 2303_CalculateAmountPaidInTaxes/2303_CalculateAmountPaidinTaxes.js
/**
 * @param {number[][]} brackets
 * @param {number} income
 * @return {number}
 */
var calculateTax = function (brackets, income) {
  let ans = 0;
  let lastUp = 0;
  for (let i = 0; i < brackets.length; i++) {
    if (income >= brackets[i][0]) {
      ans += ((brackets[i][0] - lastUp) * brackets[i][1]) / 100;
      lastUp = brackets[i][0];
    } else {
      ans += ((income - lastUp) * brackets[i][1]) / 100;
      break;
    }
  }
  return ans;
};

File: 2303_CalculateAmountPaidInTaxes/2303_CalculateAmountPaidInTaxes.py
class Solution:
    def calculateTax(self, brackets: List[List[int]], income: int) -> float:
        taxesPaid = 0
        lastbrackets = 0
        for i in range(len(brackets)):
            #destructure in python
            x, y = brackets[i]
            if income >= x:
                taxesPaid += (x - lastbrackets)*y/100
                lastbrackets = x
            else:
                taxesPaid += (income - lastbrackets)*y/100
                break
        return taxesPaid
